src/RTE.tsx

```diff
@@ -1,63 +1,98 @@
-  KeyboardEvent,
   ReactElement,
+  HTMLAttributes
 } from "react";
-import ContentEditable from "react-contenteditable";
 import Toolbar from "./components/Toolbar";
-import createAPI from "./lib/api";
-import {
-  cloneNodeAndRange,
-  getSelectionRange,
-  insertNewLine,
-  insertText,
-  isSelectionInside,
-  replaceNodeChildren,
-  replaceSelection,
-  selectEndOfNode,
-  traverse,
-} from "./lib/dom";
-import Undo from "./lib/undo";
-import styles from "./RTE.css";
+import styles from "./RTE.module.css";
+import { getBrowserInfo } from "./lib/browserInfo";
 
 export interface Feature {
-  onEnter?: (node: Node) => boolean;
-  onShortcut?: (e: KeyboardEvent) => boolean;
-  isActive?: () => boolean;
-  isDisabled?: () => boolean;
-  onSelectionChange?: () => void;
-  getFeatureInstance?: () => Feature;
+  onPathChange?: () => void;
+  onContentEditableFocus?: () => void;
+  onContentEditableBlur?: () => void;
 }
 
 interface PropTypes {
-  features?: Array<ReactElement<any>>;
-  inputId?: string;
-  onChange?: (data: { text: string; html: string }) => void;
+  /** features is an array of RTE features to be included */
+  features?: ReactElement<any>[];
+  /** inputID is the id attached to the contenteditable field */
+  inputID?: string;
+  /** onChange is called whenenver the `html` value has changed */
+  onChange?: (html: string) => void;
+  /** disabled lets you turn on/off the RTE */
   disabled?: boolean;
+  /** className added to the root */
   className?: string;
+  /** className added to the root when disabled */
   classNameDisabled?: string;
+  /** className added to the html content */
   contentClassName?: string;
+  /** className added to the html content when disabled */
   contentClassNameDisabled?: string;
+  /** className added to the html content container */
   contentContainerClassName?: string;
+  /** className added to the html content container when disabled */
   contentContainerClassNameDisabled?: string;
+  /** className added to the toolbar */
   toolbarClassName?: string;
+  /** className added to the toolbar when disabled */
   toolbarClassNameDisabled?: string;
+  /** className added to the placeholder */
   placeholderClassName?: string;
+  /** className added to the placeholder when disabled */
   placeholderClassNameDisabled?: string;
+  /** placeholder to show when RTE is empty */
   placeholder?: string;
+  /** current html value */
   value?: string;
+  /** toolbarPosition lets you switch the toolbar to top/bottom */
   toolbarPosition?: "top" | "bottom";
+  /** onFocus is called whenenver the RTE receives focus */
   onFocus?: EventHandler<FocusEvent>;
+  /** onFocus is called whenenver the RTE looses focus */
   onBlur?: EventHandler<FocusEvent>;
+  /** Only allow pasting text */
+  pasteTextOnly?: boolean;
+  /**
+   * If true, don't sanitize when `value` is applied.
+   * Important: No sanitization will take place unless `sanitizeToDOMFragment` is set.
+   */
+  noSanitizeValue?: boolean;
```

Comment:
the naming of these is a bit inconsistent in that they're negatives... could it be `sanitizeValue` and by default set it to false?

Reply:
You mean default to `true`. Yeah makes sense!

Reply:
I removed  `noSanitizeInsert` cause it should never be changed for security reasons and renamed  `noSanitizeValue` to `sanitizeValue`. Thanks @tessalt!